### DISCUSSION TITLE
Using TypeScript for Defining Custom Operations

### DISCUSSION THREAD

**Discussion by texttechne** [SAP/open-ux-odata#707] [Q&A]
Hi,

the [documentation for the EntityContainerAPI](https://github.com/SAP/open-ux-odata/blob/main/docs/EntityContainerAPI.md) states:
> When defining your own unbound actions, you can provide a javascript (or Typescript) file with the name EntityContainer.js that will implement various actions.

Using the `EntityContainer.js` works, however when using `EntityContainer.ts` the MockServer won't pick that up.
Example:
```
import { MockDataContributor } from "@sap-ux/fe-mockserver-core";

const mock: MockDataContributor<{}> = {
  executeAction: async function (actionDefinition, actionData, keys, odataRequest) {
    switch (actionDefinition.name) {
      case "Confirm":
        console.log("hi!!!");
        return {};
    }
  },
};

export default mock;
```

My setup is a bit special as I'm using the mock server standalone, i.e. without any UI5 app. I can provide this setup additionally, but my question is: Who is responsible for TS support? Are there any examples for this kind of setup or should it work out-of-the-box?

**Comment by nlunets** [comment]
Hello @texttechne 

We support typescript, but you have to transpile the file yourself before executing the code.
So you'd have to run tsc and then run your server.

Another option would be that we transpile on the fly but so far we do not provide this functionality. If you think this is meaningful we can create an issue for it :)
